### DISCUSSION TITLE
Get micropython.mem_info() data into a string

### DISCUSSION THREAD

**Discussion by yamenk-gribaudo** [micropython/micropython#13257] [ESP32] [answered]
Hey, I would like to monitor the memory usage of my program, and a good way to do that would be to have the free memory and max free size. I can see those using mem_info, but I couldnt find a way to log that information to a file or something like that. Is it posible?
Thanks,

**Comment by yamenk-gribaudo** [comment, marked as answer]
Ok, I did this:
```
import io
import os
import micropython

buff = io.StringIO()
os.dupterm(buff)
micropython.mem_info()

buff.seek(0)
a = buff.read().strip()

print(a)
```

**Reply by Josverl** [reply]
Also See https://github.com/Josverl/micropython-magic/blob/main/samples%2Fmem_info.ipynb and other samples for ways of tracking and visualizing the memory allocation and gc over the lifetime

**Comment by Josverl** [comment]
Also note that, depending on the port ESP32 , mem_free does only report the memory allocated to micropython. RTOS may have additional memory that is not yet visible to micropython

See https://docs.micropython.org/en/latest/library/esp32.html#esp32.idf_heap_info
